numpy/core/src/umath/ufunc_object.c

```diff
@@ -3511,7 +3545,8 @@ PyUFunc_Reduceat(PyUFuncObject *ufunc, PyArrayObject *arr, PyArrayObject *ind,
 
     PyArray_Descr *descrs[3];
     PyArrayMethodObject *ufuncimpl = reducelike_promote_and_resolve(ufunc,
-            arr, out, signature, NPY_TRUE, descrs, "reduceat");
+            arr, out, signature, NPY_TRUE, descrs, NPY_UNSAFE_CASTING,
```

Comment:
What does the choice of `NPY_UNSAFE_CASTING` do here? Is it likely to change?

Reply:
It keeps the current default.  Yes, I think it would be nice to change eventually.  IIRC the main reason for "unsafe" was probably that logic ufuncs needed it, but I have a different solution for them now.